tinker_cookbook/renderers.py

```diff
@@ -606,7 +606,7 @@ def render_message(self, idx: int, message: Message, is_last: bool = False) -> R
             # This matches how Qwen3 models were trained - they only see their own thinking
             # during the current turn, not from previous turns.
             ac_content = ac_content.split("</think>")[1].lstrip()
-        elif message["role"] == "assistant" and "<think>" not in ac_content:
+        elif message["role"] == "assistant" and "<think>" not in ac_content and is_last:
             # Matching the paper, we force the assistant to start with <think>. Some SFT datasets include
             # <think> in the assistant messages, we so don't need to re-add it in those cases.
             ob_str += "<think>\n"
```

Comment:
the renderer test caught another bug:
in multiturn, `<think>` was added to the intermediate steps

```
Cookbook string: <|im_start|>user
Hello, how are you?<|im_end|>
<|im_start|>assistant
<think> <--- THIS SHOULDN'T GET ADDED
I'm fine, thank you!<|im_end|>
<|im_start|>user
What is the capital of France?<|im_end|>
<|im_start|>assistant
<think>
```